### PR TITLE
Deserializing fix

### PR DESCRIPTION
**Pull request by EmilLa1**
## Purpose

Retaining the `Payload` when deserializing a transaction from bytes.

This makes deserializing safer as one can now guard via the `TransactionType` before casting the payload to the concrete type.

Also fixed a bug where the maximum energy allowed to spend on the transaction (for smart contract related transactions) was actually set higher than what the user set. In practice the cost of checking signatures was added to what the user had specified)

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/transactions/RawPayload.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * The bytes of the raw payload.
+     */
+
+    @Getter
+    private final byte[] rawBytes;
+
+    public RawPayload(final byte[] bytes) {
+        this.rawBytes = bytes;
+    }
```
Comment (abizjak) [inline concordium-sdk/src/main/java/com/concordium/sdk/transactions/RawPayload.java:23]: Shouldn't you be checking here that there is at least 1 byte for the tag? And ideally that it's a valid tag?

File: concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java [hunk 2]
```diff
@@ -40,4 +40,13 @@ public enum TransactionType {
     TransactionType(byte type) {
         this.value = type;
     }
+
+    public static TransactionType parse(byte type) {
+        for (TransactionType transactionType : values()) {
+            if (transactionType.value == type) {
+                return transactionType;
+            }
+        }
```
Comment (abizjak) [inline concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java:49]: This makes me a bit sad :( 

Iterating through all the values to find the one we want when we in principle know the tags.

File: concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java [hunk 3]
```diff
@@ -40,4 +40,13 @@ public enum TransactionType {
...
+        throw new IllegalArgumentException("Unknown transaction type tag");
```
Comment (abizjak) [inline concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java:50]: Probably useful to add the `byte` to this exception for easier debugging.

**Review by abizjak — APPROVED** [review]

File: concordium-sdk/src/main/java/com/concordium/sdk/transactions/RawPayload.java (hunk 1 above)
Comment (EmilLa1) [reply concordium-sdk/src/main/java/com/concordium/sdk/transactions/RawPayload.java:23]: Good idea.

File: concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java (hunk 3 above)
Comment (EmilLa1) [reply concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java:50]: Good idea.

File: concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java (hunk 2 above)
Comment (EmilLa1) [reply concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransactionType.java:49]: me too  :(

**Pull request merged**